main.cpp

```diff
@@ -63,18 +63,27 @@ class IntersectionController
   {
     northSouth.setSignal(true); //Initially, North and Southbound "go"
     eastWest.setSignal(false);
+    northSouth.addObserver(this);
   }
 
   void goNorthSouth()
   {
     northSouth.setSignal(true);
     eastWest.setSignal(false);
+    northSouth.notifyObservers();
   }
 
   void goEastWest()
   {
     eastWest.setSignal(true);
     northSouth.setSignal(false);
+    northSouth.notifyObservers();
+  }
+
+  void update()
+  {
+    northbound.showSignal("North and South",northSouth.signalGo());
```

Comment:
(if you got another comment on a previous line, you may disregard it. I misunderstood what you were doing and my comment was misleading)